app/scripts/models/flow.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent the somewhat nebulous concept of an
+ * "event flow". A single flow could be a user connecting a
+ * new device to their Sync account, for instance.
+ *
+ * Each flow has two properties.
+ *
+ *   * flowId: 32-byte identifier hex-encoded as a string.
+ *   * flowBeginTime: server-generated timestamp.
+ *
+ * When a new flow is begun, it should result in an activity
+ * event being logged by the server. To that end, we push a
+ * a `flow.begin` onto the metrics event array.
+ *
+ * There is some mess surrounding when to begin a new flow,
+ * which necessitates equivalent mess in this model. It runs
+ * something like this:
+ *
+ *   1. A user arriving at /signin or /signup constitutes the
+ *      beginning of a flow, so a corresponding `flow.begin`
+ *      event must be logged.
+ *
+ *   2. A user refreshing the page on /signin or /signup does
+ *      not constitute a new flow, so events must not be logged
+ *      in those cases. This is achieved by saving the flowId
+ *      in local storage.
```

Comment:
In this case I think we *want* to skew the completion percentage, because to me it feels like two separate signin attempts.  The "sits there refreshing the page for whatever reason" case is a lot less clear-cut...but I think it's probably more accurate to count those as separate flow.begin events.

On the whole, I think I'd rather start out overcounting in these situations (so that when we iterate on the details, our metrics improve for free) than undercounting them (where making the counting more accurate would make our metrics look like they're getting worse)...